### PR TITLE
feat(CIS): Order Advanced Certificate

### PR DESCRIPTION
**Pull request by arjunchauhanibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by arjunchauhanibm** [comment]
Testing resource order advanced certificate

image

image

**Comment by arjunchauhanibm** [comment]
Testing resource origin certificate

image

image

image

**Comment by arjunchauhanibm** [comment]
Testing data origin certificate

image

image

**Comment by arjunchauhanibm** [comment]
testing import for origin certificate

image

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+)
+
+func DataSourceIBMCISOriginCertificateOrder() *schema.Resource {
+	return &schema.Resource{
+		Read:     DataIBMCISOriginCertificateRead,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS object id or CRN",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:25]: Capitalize ID in this description

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisOriginCertificateID: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:38]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+			},
+			cisOriginCertificateList: {
+				Type:        schema.TypeList,
+				Description: "List of certificate",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						cisOriginCertificateID: {
+							Type:        schema.TypeString,
+							Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:49]:
```suggestion
							Description: "Certificate ID",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+							Computed:    true,
+						},
+						cisOriginCertificateType: {
+							Type:        schema.TypeString,
+							Description: "certificate type",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:54]:
```suggestion
							Description: "Certificate type",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+						},
+						cisOriginCertificateHosts: {
+							Type:        schema.TypeList,
+							Description: "Hosts which certificate need to be ordered",
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						cisOriginCertificateValidityDays: {
+							Type:        schema.TypeInt,
+							Description: "validity days",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:65]:
```suggestion
							Description: "Validity days",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+							Computed:    true,
+						},
+						cisOriginCertificateCSR: {
+							Type:        schema.TypeString,
+							Description: "validity days",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:70]:
```suggestion
							Description: "Validity days",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+							Computed:    true,
+						},
+						cisOriginCertificatePrivateKey: {
+							Type:        schema.TypeString,
+							Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:75]:
```suggestion
							Description: "Certificate ID",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+							Computed:    true,
+						},
+						cisOriginCertificate: {
+							Type:        schema.TypeString,
+							Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:80]:
```suggestion
							Description: "Certificate ID",
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+							Computed:    true,
+						},
+						cisOriginCertificateExpiresOn: {
+							Type:        schema.TypeString,
+							Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:85]:
```suggestion
							Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+	return &schema.Resource{
+		Create:   ResourceIBMCISAdvancedCertificatePackOrderCreate,
+		Update:   ResourceIBMCISAdvancedCertificatePackOrderRead,
+		Read:     ResourceIBMCISAdvancedCertificatePackOrderRead,
+		Delete:   ResourceIBMCISAdvancedCertificatePackOrderDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS object id or CRN",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:41]:
```suggestion
				Description: "CIS object ID or CRN",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisAdvancedCertificatePackOrderID: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:54]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+				Computed:    true,
+			},
+			cisAdvancedCertificatePackOrderType: {
+				Type:        schema.TypeString,
+				Description: "certificate type",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:59]:
```suggestion
				Description: "Certificate type",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+				Optional:    true,
+				Default:     cisAdvancedCertificatePackOrderTypeAdvanced,
+				ValidateFunc: validate.InvokeValidator(ibmCISAdvancedCertificatePackOrder,
+					cisAdvancedCertificatePackOrderType),
+			},
+			cisAdvancedCertificatePackOrderHosts: {
+				Type:        schema.TypeList,
+				Description: "Hosts which certificate need to be ordered",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:67]:
```suggestion
				Description: "Hosts for which certificates need to be ordered",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			cisAdvancedCertificatePackOrderStatus: {
+				Type:        schema.TypeString,
+				Description: "certificate status",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:73]:
```suggestion
				Description: "Certificate status",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+				Computed:    true,
+			},
+			cisAdvancedCertificatePackValidationMethod: {
+				Type:        schema.TypeString,
+				Description: "validation method",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:78]:
```suggestion
				Description: "Validation method",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 16]
```diff
@@ -0,0 +1,189 @@
...
+				Required:    true,
+			},
+			cisAdvancedCertificatePackValidityDays: {
+				Type:        schema.TypeInt,
+				Description: "validity days",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:83]:
```suggestion
				Description: "Validity days",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 17]
```diff
@@ -0,0 +1,189 @@
...
+				Required:    true,
+			},
+			cisAdvancedCertificatePackCertificateAthority: {
+				Type:        schema.TypeString,
+				Description: "certificate authority",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:88]:
```suggestion
				Description: "Certificate authority",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 18]
```diff
@@ -0,0 +1,189 @@
...
+				Required:    true,
+			},
+			cisAdvancedCertificatePackCloudflareBranding: {
+				Type:        schema.TypeBool,
+				Description: "cloudflare branding",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:93]:
```suggestion
				Description: "Cloudflare branding",
```

File: ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+func ResourceIBMCISAdvancedCertificatePackOrderDelete(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(conns.ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+	certificateID, zoneID, crn, err := flex.ConvertTfToCisThreeVar(d.Id())
+	if err != nil {
+		log.Println("Error in reading certificate id")
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_advanced_certificate_pack_order.go:176]:
```suggestion
		log.Println("Error in reading certificate ID")
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 20]
```diff
@@ -0,0 +1,263 @@
...
+	return &schema.Resource{
+		Create:   ResourceIBMCISOriginCertificateCreate,
+		Update:   ResourceIBMCISOriginCertificateRead,
+		Read:     ResourceIBMCISOriginCertificateRead,
+		Delete:   ResourceIBMCISOriginCertificateDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS object id or CRN",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:38]:
```suggestion
				Description: "CIS object ID or CRN",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 21]
```diff
@@ -0,0 +1,263 @@
...
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisOriginCertificateID: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:51]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 22]
```diff
@@ -0,0 +1,263 @@
...
+				Computed:    true,
+			},
+			cisOriginCertificateType: {
+				Type:        schema.TypeString,
+				Description: "certificate type",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:56]:
```suggestion
				Description: "Certificate type",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 23]
```diff
@@ -0,0 +1,263 @@
...
+				Required:    true,
+			},
+			cisOriginCertificateHosts: {
+				Type:        schema.TypeList,
+				Description: "Hosts which certificate need to be ordered",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:61]:
```suggestion
				Description: "Hosts for which certificates need to be ordered",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 24]
```diff
@@ -0,0 +1,263 @@
...
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			cisOriginCertificateValidityDays: {
+				Type:        schema.TypeInt,
+				Description: "validity days",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:67]:
```suggestion
				Description: "Validity days",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 25]
```diff
@@ -0,0 +1,263 @@
...
+				Required:    true,
+			},
+			cisOriginCertificateCSR: {
+				Type:        schema.TypeString,
+				Description: "validity days",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:72]:
```suggestion
				Description: "Validity days",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 26]
```diff
@@ -0,0 +1,263 @@
...
+				Required:    true,
+			},
+			cisOriginCertificatePrivateKey: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:77]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 27]
```diff
@@ -0,0 +1,263 @@
...
+				Computed:    true,
+			},
+			cisOriginCertificate: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:82]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 28]
```diff
@@ -0,0 +1,263 @@
...
+				Computed:    true,
+			},
+			cisOriginCertificateExpiresOn: {
+				Type:        schema.TypeString,
+				Description: "certificate id",
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:87]:
```suggestion
				Description: "Certificate ID",
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 29]
```diff
@@ -0,0 +1,263 @@
...
+
+	opt := cisClient.NewCreateOriginCertificateOptions(crn, zoneID)
+	opt.SetHostnames(hostsList)
+	opt.SetCsr(csr)
+	opt.SetRequestType(certType)
+	opt.SetRequestedValidity(validityDays)
+
+	result, resp, err := cisClient.CreateOriginCertificate(opt)
+	if err != nil {
+		log.Printf("Origin Certificate  order failed: %v", resp)
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:135]:
```suggestion
		log.Printf("Origin Certificate order failed: %v", resp)
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 30]
```diff
@@ -0,0 +1,263 @@
...
+
+func ResourceIBMCISOriginCertificateRead(d *schema.ResourceData, meta interface{}) error {
+
+	cisClient, err := meta.(conns.ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+	certificateID, zoneID, crn, err := flex.ConvertTfToCisThreeVar(d.Id())
+	if err != nil {
+		log.Println("Error in reading certificate id")
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:151]:
```suggestion
		log.Println("Error in reading certificate ID")
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 31]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+func ResourceIBMCISOriginCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(conns.ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+	certificateID, zoneID, crn, err := flex.ConvertTfToCisThreeVar(d.Id())
+	if err != nil {
+		log.Println("Error in reading certificate id")
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:182]:
```suggestion
		log.Println("Error in reading certificate ID")
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 32]
```diff
@@ -0,0 +1,263 @@
...
+
+/*
+func ResourceIBMCISCertificateOrderExist(d *schema.ResourceData, meta interface{}) (bool, error) {
+	cisClient, err := meta.(conns.ClientSession).CisSSLClientSession()
+	if err != nil {
+		return false, err
+	}
+	certificateID, zoneID, crn, _ := flex.ConvertTfToCisThreeVar(d.Id())
+	if err != nil {
+		log.Println("Error in reading certificate id")
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:210]:
```suggestion
		log.Println("Error in reading certificate ID")
```

File: ibm/service/cis/resource_ibm_cis_origin_certificate_order.go [hunk 33]
```diff
@@ -0,0 +1,263 @@
...
+*/
+/*
+func waitForCISCertificateOrderDelete(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	cisClient, err := meta.(conns.ClientSession).CisSSLClientSession()
+	if err != nil {
+		return nil, err
+	}
+	certificateID, zoneID, crn, _ := flex.ConvertTfToCisThreeVar(d.Id())
+	if err != nil {
+		log.Println("Error in reading certificate id")
```
Comment (sheltons) [inline ibm/service/cis/resource_ibm_cis_origin_certificate_order.go:236]:
```suggestion
		log.Println("Error in reading certificate ID")
```

File: website/docs/d/cis_origin_certificates [hunk 34]
```diff
@@ -0,0 +1,41 @@
+---
+subcategory: "Internet services"
+layout: "ibm"
+page_title: "IBM: ibm_cis_origin_certificates"
+description: |-
+  Get information on IBM Cloud Internet Services Certificates.
+---
+
+# ibm_cis_origin_certificates
+Retrieve an information of an existing IBM Cloud Internet Services certificates resource. For more information about CIS origin certificate, refer to [managing origin certificates](https://cloud.ibm.com/docs/cis?topic=cis-cis-origin-certificates).
```
Comment (sheltons) [inline website/docs/d/cis_origin_certificates:10]:
```suggestion
Retrieve the information of an existing IBM Cloud Internet Services certificates resource. For more information about CIS origin certificates, refer to [managing origin certificates](https://cloud.ibm.com/docs/cis?topic=cis-cis-origin-certificates).
```

File: website/docs/d/cis_origin_certificates [hunk 35]
```diff
@@ -0,0 +1,41 @@
...
+  certificate_id = "25392180178235735583993116186144990011711092749"
+}
+```
+
+## Argument reference
+Review the argument references that you can specify for your data source. 
+
+- `cis_id` - (Required, String) The ID of the CIS instance.
+- `domain_id` - (Required, String) The ID of the domain.
+- `certificate_id` - (Optional, String) The ID of the certificate. If ID is not provided you will get the list of certificates. If ID is provided then you will get the information of the particular certificate.
```
Comment (sheltons) [inline website/docs/d/cis_origin_certificates:28]:
```suggestion
- `certificate_id` - (Optional, String) The ID of the certificate. If the ID is not provided, you will get the list of certificates. If the ID is provided, then you will get the information of that certificate.
```

File: website/docs/d/cis_origin_certificates [hunk 36]
```diff
@@ -0,0 +1,41 @@
...
+
+## Attribute reference
+In addition to all argument reference list, you can access the following attribute references after your data source is created.
```
Comment (sheltons) [inline website/docs/d/cis_origin_certificates:31]:
```suggestion
In addition to the argument reference list, you can access the following attribute references after your data source is created. 
```

File: website/docs/d/cis_origin_certificates [hunk 37]
```diff
@@ -0,0 +1,41 @@
...
+
+## Attribute reference
+In addition to all argument reference list, you can access the following attribute references after your data source is created. 
+
+- `origin_certificate_list` - (String) The collection of the certificates.
+
+origin_certificate_list
+   - `certificate_id` - (String) The certificate ID.
+   - `certificate` (String) Certificate associated with the origin certificate.
+   - `expires_on` - (String) Expiry date of the certificate.
```
Comment (sheltons) [inline website/docs/d/cis_origin_certificates:38]:
```suggestion
   - `expires_on` - (String) Expiration date of the certificate.
```

File: website/docs/r/cis_advanced_certificate_pack_order.html.markdown [hunk 38]
```diff
@@ -0,0 +1,45 @@
+---
+
+subcategory: "Internet services"
+layout: "ibm"
+page_title: "IBM: ibm_cis_advanced_certificate_pack_order"
+description: |-
+  Provides a IBM CIS certificate order resource.
```
Comment (sheltons) [inline website/docs/r/cis_advanced_certificate_pack_order.html.markdown:7]:
```suggestion
  Provides an IBM CIS certificate order resource.
```

File: website/docs/r/cis_advanced_certificate_pack_order.html.markdown [hunk 39]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+# ibm_cis_advanced_certificate_pack_order
+
+ Provides an IBM Cloud Internet Services advanced certificate order resource. This resource is associated with an IBM Cloud Internet Services instance and a CIS domain resource. It allows to order and delete dedicated advanced certificates of a domain of a CIS instance. For more information about CIS certificate order, see [managing edge certificates](https://cloud.ibm.com/docs/cis?topic=cis-managing-edge-certs).
```
Comment (sheltons) [inline website/docs/r/cis_advanced_certificate_pack_order.html.markdown:12]:
```suggestion
 Provides an IBM Cloud Internet Services advanced certificate order resource. This resource is associated with an IBM Cloud Internet Services instance and a CIS domain resource. It allows you to order and delete dedicated advanced certificates of a domain of a CIS instance. For more information about CIS certificate ordering, see [managing edge certificates](https://cloud.ibm.com/docs/cis?topic=cis-managing-edge-certs).
```

File: website/docs/r/cis_advanced_certificate_pack_order.html.markdown [hunk 40]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+## Argument reference
+Review the argument references that you can specify for your resource. 
+
+- `cis_id` - (Required, String) The ID of the IBM Cloud Internet Services instance.
+- `domain_id` - (Required, String) The ID of the domain.
+- `hosts` - (Required, String) The hosts for the certificates to be ordered.
+- `certificate_authority` - (Required, String) The certificate authority selected for the order. Allowed values are `google` and `lets_encrypt`
+- `cloudflare_branding` - (Optional, Boolean) Whether or not to add Cloudflare Branding for the order.
```
Comment (sheltons) [inline website/docs/r/cis_advanced_certificate_pack_order.html.markdown:35]:
```suggestion
- `cloudflare_branding` - (Optional, Boolean) Whether to add Cloudflare branding for the order.
```

File: website/docs/r/cis_advanced_certificate_pack_order.html.markdown [hunk 41]
```diff
@@ -0,0 +1,45 @@
...
+- `validation_method` - (Required, String) Validation methond selected for the order. Allowed values are `txt`, `http`, and `email`.
+- `validity`- (Required, Int) Validty days for the order. Allowed values are `14`, `30`, `90`, `365`.
+
+
+## Attribute reference
+In addition to all argument reference list, you can access the following attribute reference after your resource is created.
```
Comment (sheltons) [inline website/docs/r/cis_advanced_certificate_pack_order.html.markdown:41]:
```suggestion
In addition to the argument reference list, you can access the following attribute reference after your resource is created.
```

File: website/docs/r/cis_advanced_certificate_pack_order.html.markdown [hunk 42]
```diff
@@ -0,0 +1,45 @@
...
+
+- `certificate_id`- (String) The certificate ID.
+- `id` - (String) The record ID. It is a combination of `<certificate_id>,<domain_id>,<cis_id>` attributes concatenated with `:`.
```
Comment (sheltons) [inline website/docs/r/cis_advanced_certificate_pack_order.html.markdown:44]:
```suggestion
- `id` - (String) The record ID, which is a combination of `<certificate_id>,<domain_id>,<cis_id>` attributes concatenated with `:`.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 43]
```diff
@@ -0,0 +1,71 @@
+---
+
+subcategory: "Internet services"
+layout: "ibm"
+page_title: "IBM: ibm_cis_origin_certificate_order"
+description: |-
+  Provides a IBM CIS origin certificate order resource.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:7]:
```suggestion
  Provides an IBM CIS origin certificate order resource.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 44]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+# ibm_cis_origin_certificate_order
+
+ Provides an IBM Cloud Internet Services origin certificate order resource. This resource is associated with an IBM Cloud Internet Services instance and a CIS domain resource. It allows to order and delete dedicated advanced certificates of a domain of a CIS instance. For more information about CIS certificate order, see [managing origin certificates](https://cloud.ibm.com/docs/cis?topic=cis-cis-origin-certificates).
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:12]:
```suggestion
Provides an IBM Cloud Internet Services origin certificate order resource. This resource is associated with an IBM Cloud Internet Services instance and a CIS domain resource. It allows you to order and delete dedicated advanced certificates of a domain of a CIS instance. For more information about CIS certificate ordering, see [managing origin certificates](https://cloud.ibm.com/docs/cis?topic=cis-cis-origin-certificates).
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 45]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+## Argument reference
+Review the argument references that you can specify for your resource. 
+
+- `cis_id` - (Required, String) The ID of the IBM Cloud Internet Services instance.
+- `domain_id` - (Required, String) The ID of the domain.
+- `hosts` - (Required, String) The hosts for the certificates to be ordered.
+- `request_type` - (Required, String) The type of the certificate. Allowed values are `origin-rsa`, `origin-ecc
+` and `keyless-certificate`.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:36]:
```suggestion

```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 46]
```diff
@@ -0,0 +1,71 @@
...
+
+```
+
+## Argument reference
+Review the argument references that you can specify for your resource. 
+
+- `cis_id` - (Required, String) The ID of the IBM Cloud Internet Services instance.
+- `domain_id` - (Required, String) The ID of the domain.
+- `hosts` - (Required, String) The hosts for the certificates to be ordered.
+- `request_type` - (Required, String) The type of the certificate. Allowed values are `origin-rsa`, `origin-ecc
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:35]:
```suggestion
- `request_type` - (Required, String) The type of the certificate. Allowed values are `origin-rsa`, `origin-ecc` and `keyless-certificate`.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 47]
```diff
@@ -0,0 +1,71 @@
...
+` and `keyless-certificate`.
+- `requested_validity`- (Required, Int) Validty days for the order. Allowed values are `7`, `30`, `90`, `365`, `730`, `1095`, `5475`.
+- `csr` - (Required, String) The Certificate Signing Request.
+
+
+## Attribute reference
+In addition to all argument reference list, you can access the following attribute reference after your resource is created.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:42]:
```suggestion
In addition to the argument reference list, you can access the following attribute reference after your resource is created.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 48]
```diff
@@ -0,0 +1,71 @@
...
+
+- `certificate_id`- (String) The certificate ID.
+- `id` - (String) The record ID. It is a combination of `<certificate_id>,<domain_id>,<cis_id>` attributes concatenated with `:`.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:45]:
```suggestion
- `id` - (String) The record ID, which is a combination of `<certificate_id>,<domain_id>,<cis_id>` attributes concatenated with `:`.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 49]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## Import
+The `ibm_cis_origin_certificate_order` resource can be imported using the ID. The ID is formed from the certificate ID, the domain ID of the domain and the CRN  Concatenated  by using a `:` character.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:49]:
```suggestion
The `ibm_cis_origin_certificate_order` resource can be imported using the ID. The ID is formed from the certificate ID, the domain ID of the domain, and the CRN  Concatenated  by using a `:` character.
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 50]
```diff
@@ -0,0 +1,71 @@
...
+
+The domain ID and CRN is located on the **Overview** page of the IBM Cloud Internet Services instance of the console domain heading, or by using the `ibmcloud cis` command line commands.
+
+- **Domain ID** is a 32 digit character string of the form: `9caf68812ae9b3f0377fdf986751a78f`
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:53]:
```suggestion
- **Domain ID** is a 32-digit character string of the form: `9caf68812ae9b3f0377fdf986751a78f`
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 51]
```diff
@@ -0,0 +1,71 @@
...
+
+- **CRN** is a 120 digit character string of the form: `crn:v1:bluemix:public:internet-svcs:global:a/4ea1882a2d3401ed1e459979941966ea:31fa970d-51d0-4b05-893e-251cba75a7b3::`
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:55]:
```suggestion
- **CRN** is a 120-digit character string of the form: `crn:v1:bluemix:public:internet-svcs:global:a/4ea1882a2d3401ed1e459979941966ea:31fa970d-51d0-4b05-893e-251cba75a7b3::`
```

File: website/docs/r/cis_origin_certificate_order.html.markdown [hunk 52]
```diff
@@ -0,0 +1,71 @@
...
+
+- **Certificate ID** is a 48 digit character string of the form: `484582976896327736468082847548136290560450732393`.
```
Comment (sheltons) [inline website/docs/r/cis_origin_certificate_order.html.markdown:57]:
```suggestion
- **Certificate ID** is a 48-digit character string of the form: `484582976896327736468082847548136290560450732393`.
```

File: ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go (hunk 1 above)
Comment (sheltons) [inline ibm/service/cis/data_source_ibm_cis_origin_certificate_order.go:25]:
```suggestion
				Description: "CIS object ID or CRN",
```

**Review by sheltons — CHANGES_REQUESTED** [review]
I've left comments/suggestions throughout.

**Comment by hkantare** [comment]
Fix the conflict in go.sum

**Review by hkantare — APPROVED** [review]

**Review by hkantare — APPROVED** [review]

**Pull request merged**